src/main/java/edu/iu/terracotta/service/app/impl/AssignmentServiceImpl.java

```diff
@@ -81,6 +81,16 @@ public AssignmentDto toDto(Assignment assignment) {
         assignmentDto.setAssignmentOrder(assignment.getAssignmentOrder());
         assignmentDto.setExposureId(assignment.getExposure().getExposureId());
         assignmentDto.setResourceLinkId(assignment.getResourceLinkId());
+        List<Treatment> treatments = allRepositories.treatmentRepository.findByAssignment_AssignmentId(assignment.getAssignmentId());
```

Comment:
What if you search directly in the assessmentRepository by assignmentID. We save a query and a for.... and in fact... you could search with a count in the submissionsRepository by assignmentId and if 0, then false if >0 then true. All with just one query.